fix(NotificationDrawer): add Translation capability

gilad215 opened the pull request:
affects: patternfly-react

<!--
Thanks for your interest in patternfly-react. We appreciate all issues filed and PRs submitted!

Please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What issue is being addressed here?) -->
**What**:
Added ability to add Translations to the Drawer. Components that can be translated:
```
  translations: {
    title: 'Notifications',
    emptyState: 'No Notifications Available',
    readAll: 'Mark All Read',
    clearAll: 'Clear All'
  }
```
Panels and Notifications were already translatable.



<!-- feel free to add additional comments -->

cdcabrera commented on packages/core/src/components/Notification/Wrappers/NotificationDrawerPanelWrapper.js:
```diff
@@ -88,14 +89,14 @@ const NotificationDrawerPanelWrapper = ({
         >
           <Button bsStyle="link" onClick={() => onMarkPanelAsRead(panelkey)}>
-            Mark All Read
+            {translations.readAll}
```
@gilad215  just curious what happens when someone accidentally replaces the entire `translations` object instead of merging or `Object.assigning` it, or is that even a concern? If we are could a possible solution involve placing checks for the translation object property, or maybe they're required?

gilad215 replied:
`translations` is not a required prop.
I don't see how someone can accidentally override the default object without actually writing `translations={...}` 
If he makes mistakes, thats his responsibility to fix it

cdcabrera replied:
For clarity, if a consumer only handles these properties ...
```
translations: {
  emptyState: 'No Notifications Available',
  readAll: 'Mark All Read',
  clearAll: 'Clear All'
}
```
What's the expect outcome?

gilad215 replied:
The title will be empty, Do you think consumers will want to translate only some of the properties? seems weird to me. Also, having a prop for each component doesn't look like a solution either, this is a very rare case..

gilad215 commented:
- Added Spread wrapper so consumers can now translate only the properties they want to @cdcabrera

```
// Inside NotificationDrawerWrapper
  const translationsWrapper = {
    ...NotificationDrawerPanelWrapper.defaultProps.translations,
    ...translations
  };
```
 
- Added more translatable components : `unreadEvent, unreadEvents, deleteNotification`
- Added `NotificationDrawerEmptyState` with a default english title
